gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java

```diff
@@ -647,27 +657,101 @@ public Long megabytesCopiedPerChunk() {
+     * exception if target blob's content type is {@code null}.
+     *
+     * @param sourceBlobId a {@code BlobId} object for the source blob
+     * @param target a {@code BlobInfo} object for the target blob
+     * @return a copy request.
+     * @throws IllegalArgumentException if {@code target.contentType} is {@code null}
+     */
+    public static CopyRequest of(BlobId sourceBlobId, BlobInfo target)
+        throws IllegalArgumentException {
+      checkContentType(target);
       return builder().source(sourceBlobId).target(target).build();
     }
 
+    /**
+     * Creates a copy request.
+     *
+     * @param sourceBucket name of the bucket containing both the source and the target blob
+     * @param sourceBlob name of the source blob
+     * @param targetBlob name of the target blob
+     * @return a copy request.
+     */
     public static CopyRequest of(String sourceBucket, String sourceBlob, String targetBlob) {
-      return of(sourceBucket, sourceBlob,
-          BlobInfo.builder(BlobId.of(sourceBucket, targetBlob)).build());
+      return CopyRequest.builder()
+          .source(sourceBucket, sourceBlob)
+          .target(BlobId.of(sourceBucket, targetBlob))
+          .build();
+    }
+
+    /**
+     * Creates a copy request.
+     *
+     * @param sourceBucket name of the bucket containing the source blob
+     * @param sourceBlob name of the source blob
+     * @param target a {@code BlobId} object for the target blob
+     * @return a copy request.
+     */
+    public static CopyRequest of(String sourceBucket, String sourceBlob, BlobId target) {
+      return builder().source(sourceBucket, sourceBlob).target(target).build();
     }
 
+    /**
+     * Creates a copy request.
+     *
+     * @param sourceBlobId a {@code BlobId} object for the source blob
+     * @param targetBlob name of the target blob, in the same bucket of the source blob
+     * @return a copy request.
+     */
     public static CopyRequest of(BlobId sourceBlobId, String targetBlob) {
-      return of(sourceBlobId,
-          BlobInfo.builder(BlobId.of(sourceBlobId.bucket(), targetBlob)).build());
+      return CopyRequest.builder()
+          .source(sourceBlobId)
+          .target(BlobId.of(sourceBlobId.bucket(), targetBlob))
+          .build();
+    }
+
+    /**
+     * Creates a copy request.
+     *
+     * @param sourceBlobId a {@code BlobId} object for the source blob
+     * @param targetBlobId a {@code BlobId} object for the target blob
+     * @return a copy request.
+     */
+    public static CopyRequest of(BlobId sourceBlobId, BlobId targetBlobId) {
+      return CopyRequest.builder()
+          .source(sourceBlobId)
+          .target(targetBlobId)
+          .build();
     }
 
     public static Builder builder() {
       return new Builder();
     }
+
+    private static void checkContentType(BlobInfo blobInfo) throws IllegalArgumentException {
+      if (blobInfo.contentType() == null) {
```

Comment:
`Preconditions.checkArgument(blobInfo.contentType() != null, "Blob content type can not be null");`?